module/input/Camera/src/Camera.cpp

```diff
@@ -462,12 +228,408 @@ namespace module::input {
+
+        // Load Hpc from configuration
+        cam.Hpc = Eigen::Matrix4d(config["lens"]["Hpc"].as<Expression>());
+
+        // Apply image offsets to lens_centre, optical axis:
+        int full_width  = arv::device_get_integer_feature_value(device.get(), "WidthMax");
+        int full_height = arv::device_get_integer_feature_value(device.get(), "HeightMax");
+
+        int offset_x = int(config["settings"]["OffsetX"].as<Expression>());
+        int offset_y = int(config["settings"]["OffsetY"].as<Expression>());
+        int width    = int(config["settings"]["Width"].as<Expression>());
+        int height   = int(config["settings"]["Height"].as<Expression>());
+
+        // Renormalise the focal length
+        float focal_length = float(config["lens"]["focal_length"].as<Expression>() * full_width / width);
+        float fov          = float(config["lens"]["fov"].as<Expression>());
+
+        // Recentre/renormalise the centre
+        Eigen::Vector2f centre =
+            (Eigen::Vector2f(config["lens"]["centre"].as<Expression>()) * full_width
+             + Eigen::Vector2f(offset_x - (full_width - width) * 0.5, offset_y - (full_height - height) * 0.5))
+            / width;
+
+        // Adjust the distortion parameters for the new width units
+        Eigen::Vector2f k = config["lens"]["k"].as<Expression>();
+        k[0]              = k[0] * std::pow(width / full_width, 2);
+        k[1]              = k[1] * std::pow(width / full_width, 4);
+
+        // Set the lens parameters from configuration
+        cam.lens = Image::Lens{
+            config["lens"]["projection"].as<std::string>(),
+            focal_length,
+            fov,
+            centre,
+            k,
+        };
+
+        // If the lens fov was auto we need to correct it
+        if (!std::isfinite(cam.lens.fov)) {
+            const double a = double(height) / double(width);
+            std::array<double, 4> options{
+                utility::vision::unproject(Eigen::Vector2f(0, 0), cam.lens, Eigen::Vector2f(1, a)).x(),
+                utility::vision::unproject(Eigen::Vector2f(1, 0), cam.lens, Eigen::Vector2f(1, a)).x(),
+                utility::vision::unproject(Eigen::Vector2f(0, a), cam.lens, Eigen::Vector2f(1, a)).x(),
+                utility::vision::unproject(Eigen::Vector2f(1, a), cam.lens, Eigen::Vector2f(1, a)).x()};
+            cam.lens.fov = float(std::acos(*std::min_element(options.begin(), options.end())) * 2.0);
+        }
+
+        // Apply the region to the camera
+        arv::camera_set_region(camera.get(), offset_x, offset_y, width, height);
+
+        // Go through our settings and apply them to the camera
+        for (const auto& cfg : config["settings"]) {
+            auto key = cfg.first.as<std::string>();
+
+            // Skip the region keys as we handle them above
+            if (key == "Width" || key == "Height" || key == "OffsetX" || key == "OffsetY" || key == "PacketSize") {
```

Comment:
I don't see `PacketSize` handled above?

Reply:
True... that one is handled below.
It's for gige vision cameras

Reply:
Please update the comments to reflect this.